aten/src/ATen/SparseCsrTensorImpl.cpp

```diff
@@ -59,16 +60,12 @@ SparseCsrTensorImpl::SparseCsrTensorImpl(
 void SparseCsrTensorImpl::resize_and_clear_(
     const int64_t nnz_size,
     IntArrayRef size) {
-  // call crow_indices().options() here since the struct contructor calls the
-  // tensor constructor with args for device specific init.
-  auto empty_crow_indices = at::empty(size[0] + 1, crow_indices().options());
-  auto empty_col_indices = at::empty(nnz_size, col_indices().options());
-  auto empty_values = at::empty(nnz_size, values().options());
+  at::native::resize_output(crow_indices_, size[0] + 1);
+  at::native::resize_output(col_indices_, nnz_size);
+  at::native::resize_output(values_, nnz_size);
```

Comment:
...should this have been stacked on https://github.com/pytorch/pytorch/pull/56744

Reply:
These `resize_output` calls are no-ops considering that `set_member_tensors` will reset the members:
https://github.com/pytorch/pytorch/blob/05a4b163a0deef2a9e9ba43594628e7346ae86c2/aten/src/ATen/SparseCsrTensorImpl.cpp#L134-L136

When creating a new CSR tensor in `_sparse_csr_tensor_unsafe`, can we eliminate the call to `resize_and_clear_` (for a new tensor there would be nothing to resize nor clear, and `set_members_tensors` will reset the members anyway) and just do the `sizes_and_strides_` and `numel` updates explicitly?

Reply:
I agree with the idea to just do the sizes_and_strides_ and numel.  So I changed this and remove `resize_and_clear_`.